### DISCUSSION TITLE
:prConcurrentLimitNone/:prHourlyLimitNone default to 99, not unlimited

### DISCUSSION THREAD

**Discussion by fgreinacher** [renovatebot/renovate#24598] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

We are running Renovate against a lot of repositories and update most dependencies on Sunday morning. That means we usually get a lot of MRs at the same time. We therefore use the `:prConcurrentLimitNone`/`:prHourlyLimitNone` presets to get all updates (as per the documentation: "Removes rate limit for PR creation per hour."). 

Looking at the implementation this is not fully correct (`if (config.prHourlyLimit)` will evaluate to `false` and the method return `99`): 

https://github.com/renovatebot/renovate/blob/bb6bc356dcae5753197a093a99466f043b07b8bd/lib/workers/repository/process/limits.ts#L12-L39

Either the documentation or the implementation needs a fix here I'd say :)

Let me know what you think!

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
Do you mean that it should be unlimited and not 99? I think we had 99 as a kind of sanity limit at the time

**Reply by fgreinacher** [reply]
Yes, as the docs say: 

- ["Removes rate limit for PR creation per hour."](https://docs.renovatebot.com/presets-default/#prhourlylimitnone) and
- ["Limit to a maximum of x concurrent branches/PRs. 0 means no limit."](https://docs.renovatebot.com/configuration-options/#prconcurrentlimit)

**Reply by rarkins** [reply]
I don't mind if you submit a PR with suitably large value so that it's effectively unlimited

![image](https://github.com/renovatebot/renovate/assets/6311784/53696124-cf22-4e68-ae07-51d5182958ea)

**Reply by fgreinacher** [reply]
🤣 We'll try with a higher number over the weekend and see if that fixes our issues.

**Reply by fgreinacher** [reply]
Maybe I'm understanding it wrong. Do these limits apply per repository or per Renovate run?

**Reply by rarkins** [reply]
Repository

**Reply by fgreinacher** [reply]
I see, that was not fully clear to me. Then it's probably not the root cause for our issue. I'll still send a PR next week.

**Comment by fgreinacher** [comment, marked as answer]
This is fixed with https://github.com/renovatebot/renovate/pull/24705
